### PR TITLE
Change password restriction and add flashes for invalid signup/login

### PR DESCRIPTION
**Pull request by jj49411**

### CODE REVIEW

File: spec/features/sign_up_spec.rb [hunk 1]
```diff
@@ -10,4 +10,26 @@
     click_button "Sign up"
     expect(page).to have_content("New post")
   end
+
+  scenario "When a registered user tries to sign up" do
+    signup
+    click_on "Logout"
+    visit "/signup"
+    fill_in "Name", with: "Ben"
+    fill_in "Email", with: "ben@gmail.com"
```
Comment (even13) [inline spec/features/sign_up_spec.rb]: @gmail => @example ?

**Review by even13 — APPROVED** [review]
fine by me - just the email domain to change, but this could be changed later, so I approve 👍

**Review by Hyan18 — APPROVED** [review]

File: spec/features/sign_up_spec.rb [hunk 2]
```diff
@@ -5,9 +5,31 @@
...
+  
+  scenario "When a user enters invalid password" do
+    visit "/signup"
+    fill_in "Name", with: "Ben"
+    fill_in "Email", with: "ben@gmail.com"
+    fill_in "Password", with: "1222"
+    fill_in "Confirm Password", with: "1222"
+    click_button "Sign up"
+    expect(page).to have_content("Password must be at least 6 characters and no more than 10")
+  end
```
Comment (Hyan18) [inline spec/features/sign_up_spec.rb:34]: Could we add a test where password and confirm password don't match

File: spec/features/sign_up_spec.rb (hunk 1 above)
Comment (jj49411) [reply spec/features/sign_up_spec.rb]: Done

File: spec/features/sign_up_spec.rb (hunk 2 above)
Comment (jj49411) [reply spec/features/sign_up_spec.rb:34]: Added!

**Pull request merged**